src/track/track.cpp

```diff
@@ -780,14 +779,41 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
     for (const auto& pCue: m_cuePoints) {
         connect(pCue.get(), &Cue::updated, this, &Track::slotCueUpdated);
         // update main cue point
-        if (pCue->getType() == Cue::Type::MainCue) {
+        if (pCue->getType() == CueInfo::Type::MainCue) {
             m_record.setCuePoint(CuePosition(pCue->getPosition()));
         }
     }
     markDirtyAndUnlock(&lock);
     emit cuesUpdated();
 }
 
+void Track::setCuePoints(const QList<CueInfo>& cueInfos) {
+    QList<CuePointer> cuePoints;
+    const double sampleRateKhz = getSampleRate() / 1000.0;
+    const double millisecsToSamplesFactor = sampleRateKhz * mixxx::kEngineChannelCount;
+    VERIFY_OR_DEBUG_ASSERT(millisecsToSamplesFactor > 0) {
+        qWarning() << "Failed to calculate millisecsToSamplesFactor!";
+        return;
+    }
+
+    for (const CueInfo& cueInfo : cueInfos) {
+        CuePointer pCue(new Cue(m_record.getId()));
+        double sampleStartPosition = cueInfo.getStartPosition() * millisecsToSamplesFactor;
```

Comment:
Cue always have start positions, right?

Reply:
No. Doesn't the Outro cue only has an end position? Management of cue positions and ranges with their different interpretation of all possible combinations is a separate topic. Just assume here that every combination is possible.